kafka-4866: Kafka console consumer property is ignored

huxihx opened the pull request:
Added `print.value` config in ConsoleConsumer to match what the quickstart document says.

huxihx committed:
kafka-4866: Kafka console consumer property is ignored

Added `print.value` config in ConsoleConsumer to match what the quickstart document says.

huxihx commented:
@ijuma a minor improvement. Please review. Thanks.

ijuma commented:
Thanks for the PR, can we add a test please?

huxihx committed:
kafka_4866: Kafka console consumer property is ignored

Add test in ConsoleConsumerTest#shouldParseValidNewSimpleConsumerValidConfigWithStringOffset for explicitly setting `print.value` to false during issuing ConsoleConsumer command

huxihx commented:
@ijuma sorry for the delay. Added a test item in `shouldParseValidNewSimpleConsumerValidConfigWithStringOffset` method for explicitly setting `print.value` to false. Could you review again? Thanks.

huxihx commented:
@ijuma Do you have any chances to review the PR?

huxihx commented:
@hachikuji Could you help review this PR? Thanks

ijuma commented on core/src/main/scala/kafka/tools/ConsoleConsumer.scala:
```diff
@@ -455,7 +458,11 @@ class DefaultMessageFormatter extends MessageFormatter {
+    if (printValue) {
+      write(valueDeserializer, value, lineSeparator)
+    } else {
+      write(None, "".getBytes(), lineSeparator)
```
Isn't it better to do `output.write(lineSeparator)` instead?

huxihx committed:
kafka-4866: Kafka console consumer property is ignored

Addressed Ijuma's comments, outputing a lineSeparator instead of invoking 'write'

huxihx commented:
@ijuma Sounds good. Already addressed your comments. Please review again.

ijuma commented on core/src/main/scala/kafka/tools/ConsoleConsumer.scala:
```diff
@@ -455,7 +458,11 @@ class DefaultMessageFormatter extends MessageFormatter {
     }
 
     if (printKey) write(keyDeserializer, key, keySeparator)
```
Maybe we should not output `keySeparator` is we are not printing values?

huxihx committed:
kafka-4866: Kafka console consumer property is ignored

Added a helper function `outputSeparator` to cover all the seven situation where printTimestamp, printKey and printValue are set with different values.

huxihx commented:
@ijuma Added a helper function `outputSeparator` to cover all the 7 situations where `printTimestamp`, `printKey` and `printValue` are set with different values.

huxihx committed:
resolved the conflicts with trunk code

huxihx committed:
addressed the conflicts

huxi-2b committed:
Merge branch 'trunk' into kafka4866_consoleconsumer_ignore_printvalue

huxihx commented:
@ijuma Did you get any chance to review the new merged code? Thanks in advance.

ijuma commented:
Thanks for the PR, LGTM. Merged to trunk after some minor tweaks.
